layouts/partials/script.html

```diff
@@ -1,106 +1,61 @@
+        {{- $scripts = union $scripts (slice "js/nav-toggle.js") -}}
     {{- end -}}
 {{- end -}}
 
 {{- if .Site.Params.enableBackToTopAutoHide -}}
-    {{- .Scratch.Add "script" (slice $backToTop) -}}
+    {{- $scripts = union $scripts (slice "js/back-to-top.js") -}}
 {{- end -}}
 
 {{- if .Site.Params.enableDarkMode -}}
-    {{- .Scratch.Add "script" (slice $darkMode) -}}
+    {{- $scripts = union $scripts (slice "js/dark-mode.js") -}}
 {{- end -}}
 
 {{- if and .Site.IsMultiLingual .Site.Params.enableLangToggle -}}
-    {{- .Scratch.Add "script" (slice $lang) -}}
+    {{- $scripts = union $scripts (slice "js/multilingual.js") -}}
 {{- end -}}
 
-{{- .Scratch.Add "script" (slice $wrapTable) -}}
+{{- $scripts = union $scripts (slice "js/wrap-table.js") -}}
 
 {{- if .Site.Params.enableCopy -}}
-    {{- .Scratch.Add "script" (slice $copy) -}}
+    {{- $scripts = union $scripts (slice "js/copy.js") -}}
 {{- end -}}
 
 {{- if and .Site.Params.enableComments (eq hugo.Environment "production") -}}
-    {{- .Scratch.Add "script" (slice $comments) -}}
+    {{- $scripts = union $scripts (slice "js/comments.js") -}}
 {{- end -}}
 
-{{ if .Site.Params.enableSmoothScroll }}
-    {{ partial "third-party/smooth-scroll.html" . }}
-{{ end }}
-
-{{ if .Params.katex | default .Site.Params.enableKaTeX }}
-    {{ partial "third-party/katex.html" . }}
-{{ end }}
-
-{{ if .Params.mathjax | default .Site.Params.enableMathJax }}
-    {{ partial "third-party/mathjax.html" . }}
-{{ end }}
-
-{{ if .Params.mermaid | default .Site.Params.enableMermaid }}
-    {{ partial "third-party/mermaid.html" . }}
-{{ end }}
-
-{{ if and (.Params.comments | default .Site.Params.enableComments) (eq hugo.Environment "production") }}
-    {{ if or (in .Site.Params.mainSections .Section) .Params.comments }}
-
-        {{ if .Site.Params.enableDisqus }}
-            {{ partial "third-party/disqus.html" . }}
-        {{ end }}
-
-        {{ if .Site.Params.enableValine }}
-            {{ partial "third-party/valine.html" . }}
-        {{ end }}
-
-        {{ if .Site.Params.enableUtterances }}
-            {{ partial "third-party/utterances.html" . }}
-        {{ end }}
-
-    {{ end }}
-{{ end }}
-
-{{ if .Site.Params.enableMediumZoom }}
-    {{ partial "third-party/medium-zoom.html" . }}
-{{ end }}
-
-{{ if .Site.Params.enableInstantPage }}
-    {{ partial "third-party/instant-page.html" . }}
-{{ end }}
-
-{{ if .Site.Params.enableLunrSearch }}
-    {{ partial "third-party/lunr-search.html" . }}
-{{ end }}
-
-{{ partial "third-party/service-worker.html" . }}
-
-{{ partial "third-party/busuanzi.html" . }}
+{{- if .Site.Params.enableLunrSearch -}}
+    {{- $scripts = union $scripts (partial "third-party/lunr-search.html" .) -}}
+{{- end -}}
 
-{{ partial "custom/script.html" . }}
+{{- $scripts = union $scripts (slice "js/custom.js") -}}
 
-{{- .Scratch.Add "script" (slice $custom) -}}
+{{- $processedScripts := slice ("" | resources.FromString "dummy.js") -}}
```

Comment:
This is a weird but it cannot be an empty slice. Apparently, slices are typed and an empty slice will be created with the wrong type making `resources.Concat` fail.